### DISCUSSION TITLE
Unit testing

### DISCUSSION THREAD

**Discussion by Pirulax** [multitheftauto/mtasa-blue#2190] [Ideas]
I propose to require some kind of unit tests for new PRs.
The main blocker of merging PRs fast right now (IMHO) is the lack of trust, and the fear of breaking something.
Properly written unit tests can filter these errors quite effectively.

What do you think?
Which library should we use?
How should we deal with Lua code? Use a resource (Maybe along a Lua unit test library?)?

**Comment by ricksterhd123** [comment]
What about small scale PRs which at minimum require Lua unit testing would it be preferable they also need unit testing in C++? I think it's a good idea and it would be good practice for a project of this scale. I had a quick search and busted looks good.

**Comment by Disinterpreter** [comment]
![image](https://user-images.githubusercontent.com/3816155/114512751-1cce7880-9c42-11eb-911b-235de55a2e8a.png)

Somebody found the idea to do it!

Totally agree with it. It can solve some of compatibility problem when somebody make a PR. But I have no ideas how to implement it 🤔 .

**Reply by Lpsd** [reply]
This is something we plan to actively review in due course - the lack of client automated tests is definitely a concern.

**Comment by alandmine** [comment]
shhhhhh little boy
